packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -236,8 +236,33 @@ export class VerifiedFetch {
 
   private async handleDagCbor ({ resource, cid, path, accept, options }: FetchHandlerFunctionArg): Promise<Response> {
     this.log.trace('fetching %c/%s', cid, path)
+    let terminalElement: ObjectNode | undefined
+    let ipfsRoots: CID[] | undefined
+
+    // need to walk path, if it exists, to get the terminal element
+    try {
+      const pathDetails = await walkPath(this.helia.blockstore, `${cid.toString()}/${path}`, options)
+      ipfsRoots = pathDetails.ipfsRoots
+      const potentialTerminalElement = pathDetails.terminalElement
+      if (potentialTerminalElement == null) {
+        return notFoundResponse(resource.toString())
```

Comment:
`resource` is already a string?

https://github.com/ipfs/helia-verified-fetch/pull/39/files#diff-12157866119b95f1b476d4e54ac9d3f22a24cc7e7af1a85f251c00c4dfbefcc6R67

```suggestion
        return notFoundResponse(resource)
```

Reply:
ah yea I see we're converting to string as part of 
`const handlerArgs: FetchHandlerFunctionArg = { resource: resource.toString(), cid, path, accept, options }` before calling codecHandlers